tutorials/plot_background_filtering.py

```diff
@@ -0,0 +1,365 @@
+# (shown to a lower limit of -60 dB for simplicity):
+
+nyq = sfreq / 2.  # the Nyquist frequency is half our sample rate
+freq = [0, f_p, f_p, nyq]
+gain = [1, 1, 0, 0]
+
+
+def box_off(ax):
+    ax.grid(zorder=0)
+    for key in ('top', 'right'):
+        ax.spines[key].set_visible(False)
+
+
+def plot_ideal(freq, gain, ax):
+    freq = np.maximum(freq, xlim[0])
+    xs, ys = list(), list()
+    for ii in range(len(freq)):
+        xs.append(freq[ii])
+        ys.append(ylim[0])
+        if ii < len(freq) - 1 and gain[ii] != gain[ii + 1]:
+            xs += [freq[ii], freq[ii + 1]]
+            ys += [ylim[1]] * 2
+    gain = 10 * np.log10(np.maximum(gain, 10 ** (ylim[0] / 10.)))
+    ax.fill_between(xs, ylim[0], ys, color='r', alpha=0.1)
+    ax.semilogx(freq, gain, 'r--', alpha=0.25, linewidth=4, zorder=3)
+    xticks = [1, 2, 4, 10, 20, 40, 100, 200, 400]
+    ax.set(xlim=xlim, ylim=ylim, xticks=xticks, xlabel='Frequency (Hz)',
+           ylabel='Amplitude (dB)')
+    ax.set(xticklabels=xticks)
+    box_off(ax)
+
+half_height = np.array(plt.rcParams['figure.figsize']) * [1, 0.5]
+ax = plt.subplots(1, figsize=half_height)[1]
+plot_ideal(freq, gain, ax)
+ax.set(title='Ideal %s Hz lowpass' % f_p)
+mne.viz.tight_layout()
+
+###############################################################################
+# This filter hypothetically achieves zero ripple in the frequency domain,
+# perfect attenuation, and perfect steepness. However, due to the discontunity
+# in the frequency response, the filter would require infinite ringing in the
+# time domain (i.e., infinite order) to be realized. Another way to think of
+# this is that a rectangular window in frequency is actually sinc_ function
+# in time, which requires an infinite number of samples, and thus infite
+# time, to represent. So although this filter has ideal frequency suppression,
+# it has poor time-domain characteristics.
+#
+# Let's try to naïvely make a brick-wall filter of length 0.1 sec:
+
+n = int(round(0.1 * sfreq)) + 1
+t = np.arange(-n // 2, n // 2) / sfreq  # center our sinc
+h = np.sinc(2 * f_p * t) / (4 * np.pi)
+
+
+def plot_filter(h, title, freq, gain):
+    fig, axs = plt.subplots(2)
+    t = np.arange(len(h)) / sfreq
+    axs[0].plot(t, h, color=blue)
+    axs[0].set(xlim=t[[0, -1]], xlabel='Time (sec)',
+               ylabel='Amplitude h(n)', title=title)
+    box_off(axs[0])
+    f, H = signal.freqz(h)
+    f *= sfreq / (2 * np.pi)
+    axs[1].semilogx(f, 10 * np.log10((H * H.conj()).real), color=blue,
+                    linewidth=2, zorder=4)
+    plot_ideal(freq, gain, axs[1])
+    mne.viz.tight_layout()
+
+plot_filter(h, 'Sinc (0.1 sec)', freq, gain)
+
+###############################################################################
+# This is not so good! Making the filter 10 times longer (1 sec) gets us a
+# bit better stop-band suppression, but still has a lot of ringing in
+# the time domain:
+
```

Comment:
... there is no signal here, unless you mean an impulse, which wouldn't add much. These are just plots of the filters themselves.